src/test/coinstatsindex_tests.cpp

```diff
@@ -79,6 +80,13 @@ BOOST_FIXTURE_TEST_CASE(coinstatsindex_initial_sync, TestChain100Setup)
     // Shutdown sequence (c.f. Shutdown() in init.cpp)
     coin_stats_index.Stop();
 
+    // Make sure coin_stats_index is not used by the scheduler thread after it
+    // has been destroyed.
+    if (m_node.scheduler) {
+        m_node.scheduler->stop();
+    }
+    GetMainSignals().FlushBackgroundCallbacks();
```

Comment:
I don't really like duplicating the shutdown sequence, especially not in the same order. This is just bloating the tests with hard to understand and hard to maintain code.

The reason that the bug exits is a presumed silent merge conflict. I already fixed the same bug in fa176e253fb473767c61d4d8cd2d93e87d71a015, but the test here copied the unfixed code.

My preference would be to use the same fix in all unit tests and not use a different fix for each unit test.

Reply:
Done.